### DISCUSSION TITLE
Data-only notification not received on react native iOS app

### DISCUSSION THREAD

**Discussion by niharikaargade** [invertase/react-native-firebase#4885] [Q&A] [answered]
Sending a data-only notification from FCM to a react-native iOS app in foreground is not received by react-native-firebase/messaging onMessage() method. However it works for an android app and for iOS if we send data + notification message. A notification-only message works completely fine on iOS and android getting this issue for data-only message.

**package.json**

    "@react-native-firebase/app": "10.0.0",
    "@react-native-firebase/crashlytics": "10.0.0",
    "@react-native-firebase/messaging": "10.0.0"

**Podfile**

      pod 'Firebase/Core'
      pod 'Firebase/Firestore'
      pod 'Firebase/Messaging'

**Payload**

       {
        "message":{
        "token":"TOKEN",
        "data":{
          "Nick" : "Mario",
          "body" : "great match!",
          "Room" : "PortugalVSDenmark"
         }
       }
     }

**App.js**

    async componentDidMount() {
            //Get notification permission from user and token
            await this.checkPermission();
            
            messaging().onMessage(async remoteMessage => {
                console.log("Remote notification inside onMessage" + JSON.stringify(remoteMessage));
            });
    
            messaging().onNotificationOpenedApp(remoteMessage => {
                console.log('Notification caused app to open from background state:', remoteMessage);
            });
    
            // Check whether an initial notification is available
            messaging().getInitialNotification().then(remoteMessage => {
                if (remoteMessage) {
                    console.log(
                        'Notification caused app to open from quit state:',
                        remoteMessage.notification,
                    );
      
                }
            });
        }

**Comment by mikehardy** [comment, marked as answer]
Please search the roughly 1,000 related data-only not coming on iOS messages in this issues list
Pay special attention to the search term "unreliable"

**Reply by niharikaargade** [reply]
Thanks for responding, I have actually read all the data-only related issues present on the discussion and  I have also come across one of your comments talking about how data-only notifications do not work when app is in background; so does that also apply to when the app is in foreground as well ? If yes, is there an efficient work around for it ? As we only need the data-only notification when the app is in foreground.

**Reply by mikehardy** [reply]
data-only definitely works in foreground, if it is not working then something is wrong with your project, either wrong entitlements or wrong payload used or wrong implementation of handlers etc. I never have a problem with foreground delivery.

**Reply by niharikaargade** [reply]
Ohh thank you so much for confirming this. I will try integrating react-native-firebase with my project again.
